pkg/controller/backup_status_updater.go

```diff
@@ -100,30 +113,219 @@ func (u *realBackupConditionUpdater) Update(backup *v1alpha1.Backup, condition *
+func updateLogSubcommandStatus(backup *v1alpha1.Backup, condition *v1alpha1.BackupCondition, newStatus *BackupUpdateStatus) bool {
+	// init log subcommand status map
+	if backup.Status.LogSubCommandStatuses == nil {
+		backup.Status.LogSubCommandStatuses = make(map[v1alpha1.LogSubCommandType]v1alpha1.LogSubCommandStatus)
+	}
+	// update subcommand status
+	subStatus, ok := backup.Status.LogSubCommandStatuses[condition.Command]
+	if !ok {
+		// init subcommand status
+		subStatus = v1alpha1.LogSubCommandStatus{
+			Command:     condition.Command,
+			TimeStarted: metav1.Time{Time: time.Now()},
+			Conditions:  make([]v1alpha1.BackupCondition, 0),
+		}
+	}
+
+	// update the status info and the condition info
+	subcommandStatusUpdate := updateLogSubCommandStatusOnly(&subStatus, newStatus)
+	subcomandConditionUpdate := updateLogSubCommandConditionOnly(&subStatus, condition)
+	if subcommandStatusUpdate || subcomandConditionUpdate {
+		backup.Status.LogSubCommandStatuses[condition.Command] = subStatus
+		return true
+	}
+	return false
+}
+
+// updateWholeLogBackupStatus updates the whole log backup status.
+func updateWholeLogBackupStatus(backup *v1alpha1.Backup, condition *v1alpha1.BackupCondition, newStatus *BackupUpdateStatus) bool {
+	var (
+		copyCondition  v1alpha1.BackupCondition
+		copyStatus     BackupUpdateStatus
+		copyConditionP *v1alpha1.BackupCondition
+		copyStatusP    *BackupUpdateStatus
+	)
+	copyCondition = *condition
+	copyConditionP = &copyCondition
+	if newStatus != nil {
+		copyStatus = *newStatus
+		copyStatusP = &copyStatus
+	}
+
+	switch condition.Command {
+	case v1alpha1.LogStartCommand:
+		// start command should not update TimeCompleted and condition when start complete.
+		// start command other status and condition info can be synced to the whole log backup status.
+		if condition.Type == v1alpha1.BackupComplete {
+			if copyStatusP != nil {
+				copyStatusP.TimeCompleted = nil
+			}
+			copyConditionP = nil
+		}
+	case v1alpha1.LogStopCommand:
+		// stop command should not update TimeStarted, but should update condition when stop complete.
+		// stop command other status should not be synced to whole log backup status.
+		if condition.Type == v1alpha1.BackupComplete {
+			if copyStatusP != nil {
+				copyStatusP.TimeStarted = nil
+			}
+		} else {
+			copyStatusP = nil
+			copyConditionP = nil
+		}
+	case v1alpha1.LogTruncateCommand:
+		// truncate command shoudld not update TimeStarted, TimeCompleted when truncate complete.
+		// truncate command other status should not be synced to whole log backup status.
+		if condition.Type == v1alpha1.BackupComplete {
+			if copyStatusP != nil {
+				copyStatusP.TimeCompleted = nil
+				copyStatusP.TimeStarted = nil
+			}
+		} else {
+			copyStatusP = nil
+		}
+		copyConditionP = nil
+	default:
+		// should not hanpen
+		return false
+	}
+
+	// update whole log status and condition
+	wholeStatusUpdate := updateBackupStatus(&backup.Status, copyStatusP)
```

Comment:
What is the purpose of code above？Values pass by pointer and reset to nil here and there, it seems not easy to understand what it trying to achieve?